mapproject.c

```diff
@@ -857,7 +858,10 @@ msProjectRectAsPolygon(projectionObj *in, projectionObj *out,
     return MS_SUCCESS;
   }
   
-  ringPoints = (pointObj*) calloc(sizeof(pointObj),NUMBER_OF_SAMPLE_POINTS*4+4);
+  /* If there is more than two sample points we will also get samples from the diagonal line */
+  sampleDiagonal = NUMBER_OF_SAMPLE_POINTS > 2 ? MS_TRUE : MS_FALSE;
```

Comment:
NUMBER_OF_SAMPLE_POINTS is a constant, so this variable isn't needed, and the below assignment can be simplified too

Reply:
That is true, I've pushed a new commit which removed redundant code.